lib/response-builder.ts

```diff
@@ -108,3 +109,23 @@ function extractDockerfileDeps(allDeps, dockerfilePkgs) {
       return extractedDeps;
     }, {});
 }
+
+function annotateLayerIds(deps, dockerfilePkgs) {
```

Comment:
why do we need `annotateLayerIds` and not using `instructionDigest` in `collectDockerfilePkgs`?

Make more sense for me...

Reply:
collectDockerfilePkgs populates dependant pkgs for the **dockerfilePackages** map, whereas annoteLayerIds annotates **depTree** with layer ids...

Reply:
was thinking about doing it here https://github.com/snyk/snyk-docker-plugin/blob/master/lib/docker-file.ts#L37

don't you think it will be much more simple?

Reply:
not possible since at this stage we don't have all the packages in dockerfilePackages.

one of the functions in response-builder flattens the packages and adds non-direct packages to dockerfilePackages, until that point dockerfilePackages only contain direct packages